### PR TITLE
Add artifacts-helper

### PR DESCRIPTION
**Pull request by RubinaUddyal**

### CODE REVIEW

**Comment by Theminijohn** [comment]
Feature request: Refresh plugin automatically on first load

File: content/artifacts/artifacts-helper/plugin.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                          });*/
+      };
+      const do_prospect = (planet) => {
+          console.log("Post P on ", planet.locationId);
+          df.prospectPlanet(planet.locationId).then(
+              v => {
+                  do_find(planet);
+              },
+              e => {
+                  console.log("Error Prospecting artifact on ", planet.locationId, " ", e);
```
Comment (gabeio) [inline content/artifacts/artifacts-helper/plugin.js:203]: [suggestion] I might replace or add next to these kinds of errors (ones that the user might need to know about) with `ui.terminal.current.print` so they are visible within the game.

**Comment by gabeio** [comment]
Going off @Theminijohn I would suggest adding an auto refresh as well as the refresh button (auto & on-demand).

**Review by gabeio — CHANGES_REQUESTED** [review]
Screen Shot 2021-06-04 at 14 21 05

The "Not ready" state doesn't seem to be functioning sadly. This foundry is not ready but is showing as waiting to prospect.

File: content/artifacts/artifacts-helper/plugin.js (hunk 1 above)
Comment (RubinaUddyal) [reply content/artifacts/artifacts-helper/plugin.js:203]: Nice! I'm quite new with this game so have no idea on the best ui design. Sorry for that.

**Comment by RubinaUddyal** [comment]
> 
> 
> Screen Shot 2021-06-04 at 14 21 05
> 
> The "Not ready" state doesn't seem to be functioning sadly. This foundry is not ready but is showing as waiting to prospect (it is not expired, it is the one that it brought me to after clicking the link).

Might write full text of "Not R" cause it's actually "Not PlanetType.RUINS" (which is useless for that it should always be zero)

**Comment by RubinaUddyal** [comment]
Any suggestions on descriptive text? To my opinion they should be short but expressive. But for now they only achieved the former goal.

File: content/artifacts/artifacts-helper/plugin.js (hunk 1 above)
Comment (gabeio) [reply content/artifacts/artifacts-helper/plugin.js]: That's all good 👍 I'm actually pretty new to it as well 😄

File: content/artifacts/artifacts-helper/plugin.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+(function (PlanetType2) {
+    PlanetType2[PlanetType2["NOT_RUINS"] = 0] = "NOT_RUINS";
+    PlanetType2[PlanetType2["NOT_PROSPECTED"] = 1] = "NOT_PROSPECTED";
+    PlanetType2[PlanetType2["PROSPECTING"] = 2] = "PROSPECTING";
+    PlanetType2[PlanetType2["READY_TO_FIND"] = 3] = "READY_TO_FIND";
+    PlanetType2[PlanetType2["FINDING"] = 4] = "FINDING";
+    PlanetType2[PlanetType2["PROSPECTED_EXPIRED"] = 5] = "PROSPECTED_EXPIRED";
+    PlanetType2[PlanetType2["NOT_READY_TO_PROSPECTED"] = 6] = "NOT_READY_TO_PROSPECTED";
+    PlanetType2[PlanetType2["FOUND"] = 7] = "FOUND";
+})(PlanetType2 || (PlanetType2 = {}));
```
Comment (phated) [inline content/artifacts/artifacts-helper/plugin.js:67]: Why not import the type from the `@darkforest_eth/types` package? That is why we published them

File: content/artifacts/artifacts-helper/plugin.js [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        });
+        this.container = null;
+        this.pluginHTMLNode = null;
+        this.chainBlockNumber = 0;
+        this.planets = [];
+    }
+
+    destroy() {
+        this._chainBlockNumber_Subscriber.unsubscribe();
+        render(null, this.container, this.pluginHTMLNode);
```
Comment (phated) [inline content/artifacts/artifacts-helper/plugin.js:180]: This changed in preact
```suggestion
        render(null, this.container);
```

File: content/artifacts/artifacts-helper/plugin.js [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+        };
+        const do_prospect = (planet) => {
+            console.log("Post P on ", planet.locationId);
+            df.prospectPlanet(planet.locationId).then(
+                v => {
+                    do_find(planet);
+                },
+                e => {
+                    console.log("Error Prospecting artifact on ", planet.locationId, " ", e);
+                });
```
Comment (phated) [inline content/artifacts/artifacts-helper/plugin.js:213]: It looks like this is always caught and never rejects

File: content/artifacts/artifacts-helper/plugin.js [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+                    do_find(planet);
+                }
+            }
+        })
+    }
+
+    async render(container) {
+        this.container = container;
+        container.style.width = "490px";
+        this.pluginHTMLNode = render(
```
Comment (phated) [inline content/artifacts/artifacts-helper/plugin.js:232]: Probably don't need to assign this anymore

File: content/artifacts/artifacts-helper/plugin.js (hunk 2 above)
Comment (RubinaUddyal) [reply content/artifacts/artifacts-helper/plugin.js:67]: I have no idea on how to import that, and even can't find the repository that generates the package on npmjs. Would you make some tips?

File: content/artifacts/artifacts-helper/plugin.js (hunk 4 above)
Comment (RubinaUddyal) [reply content/artifacts/artifacts-helper/plugin.js:213]: Do you mean I should re-throw `e` ? Is that for showing the error on console?

File: content/artifacts/artifacts-helper/plugin.js (hunk 3 above)
Comment (RubinaUddyal) [reply content/artifacts/artifacts-helper/plugin.js:180]: According to document of v10, the `replaceNode` argument still exists and not marked as deprecated?

File: content/artifacts/artifacts-helper/plugin.js (hunk 2 above)
Comment (jacobrosenthal) [reply content/artifacts/artifacts-helper/plugin.js:67]: Check out https://www.npmjs.com/package/@darkforest_eth/types Or on unpkg https://unpkg.com/@darkforest_eth/types@6.1.27/dist/index.js

**Comment by jacobrosenthal** [comment]
@RubinaUddyal you in round 2? Lets push this over the finish line.

**Comment by jacobrosenthal** [comment]
Please reopen if you come back!

**Pull request closed without merging**